### ISSUE TITLE
Don't claim to create safe spaces

### ISSUE THREAD

**Issue by nonchip** [offapi/rbac-23andme-oauth2#3]
Please don't claim to create safe spaces using the 23andme api, as it per design discriminates on trans people, people with genetic defects, races, implements kin liability, etc.

see also #1, closed by now because it's answered and no "real" issue/bugreport, while this one is: @offensive-computing you should imho mention that while this project can help filter the masses for safe spaces, it should never be the only option, because it's way too exclusive, DNA just isn't enough to filter people. you'll still get false positive trolls you'll have to moderate away, and false negative users you'll be excluding and there should be another option for them to join.

**Comment by blackfrat** [comment]
Please disregard this complaint. I am planning to create social website for black women living in underprivileged areas, and I would like to be sure my users are actually black women not white trolls  seeking to disrupt our safe community. Besides using geographic area registration access control I am very interested in this project.

**Comment by refdoc** [comment]
This is as per the example "No blacks, no Jews!". Bizarre and sick.

**Comment by blackfrat** [comment]
The examples also mention Jews only, and Blacks only websites, so you trying to turn it into "evil racist software" is laughable. 

Fortunately using immutable genetic traits as access control is the future, and it is already implemented in many places demanding high level of security. That this technology would become accessible to general public is great, and your emotional objections does not matter.

**Comment by Karunamon** [comment]
Pretty sure that second example is the textbook definition of racism.

**Comment by lleaff** [comment]
I for one look forward to my Aryan supremacy safe-space.

**Comment by nonchip** [comment]
@blackfrat: the problem is your black women dont' neccessarily have an Y chromosome, so nope, don't disregard this complain. also, genetic traits are not immutable and don't really tell you anything about the person in question. maybe a person born as a black woman undergoes therapy to change both their skin color and sexual organs and then wants to murder all your users because they joined @lleaff's safespace ;)

also, your claim of me being laughable is laughable, because I'm not trying to turn anything into evil racist software, I'm just saying that banning e.g. non-jewish/non-black/non-cis-female/etc people is by definition racism/sexism/*ism. fun fact: you don't have to be white and hate blacks to be racist, it works the other way around too (as you're demonstrating by "making sure" your users are "no white trolls").

also, genetic access control is nothing "fortunate" (because genetics don't say anything about the person, except you're using it to explicitely match one person, which your use case doesn't), and sociological facts don't have anything to do with my emotions.

I'd like to advise you to get outside and take a closer look at people: there are many cases where genetics don't only not tell you anything valuable about them, but also tell you wrong infos (see e.g. trans people trying to use your gender-filtered website).

to clarify: I don't think 23am or this api are a bad thing for managing access to individual accounts, I just think you should not filter your users by it.

also, afaik, 23am tests cost about $100, I'd really like to see the shitstorm when you tell your underprivileged users to pay that just to use your website.

**Comment by blackfrat** [comment]
@nonchip :
 First, mismatch between sex and gender identity is extremely rare and is considered to be a mental disease, it is as yet untreatable, even for people who undergone surgery to remove their body parts the suicide rate is extremely high. For 99.99% of population sex **is** immutable trait which determines gender.  I am not going to be "inclusive" for the sake of ability to effectively filter my users. 

Second, are you seriously claiming artificialy altering one's skin colour makes one a black/white/whatever person?
Furthermore, no, you can not change your sexual organs. You can have surgery to remove the organs you were born with, and have the surgeons remodel some parts of what remains to resemble organs of the opposite sex. 

 Third, you are religiously claiming to not allow anyone in is racism. This is demonstrably false. Sometimes having safe space requires exclusion of the people who would introduce hostile environment, or are not the target demographics. When I am setting up website for black underprivileged women, I am setting it for _them_ not for everyone. 

> because genetics don't say anything about the person

Yes, and pigs can fly. We have come a long since the discovery of the DNA, and here you grandly proclaim "it does not say anything about the person". Every physical trait has its reflection in the genome, including sex, skin colour, ancestry and many others. There is an ongoing research which shows links between DNA and many behavioral traits.

> also, afaik, 23am tests cost about $100, I'd really like to see the shitstorm when you tell your underprivileged users to pay that just to use your website.

We are considering obtaining government funding from one of the many programs targeted at minorities.

**Comment by nonchip** [comment]
ok, @blackfrat, first, I'd like to thank you for calling me a border case and mentally ill ;)

> First, mismatch between sex and gender identity is extremely rare

yeah, sure, 14% of the population is "extremely rare": https://en.wikipedia.org/wiki/Transsexual#Prevalence

oh, you shouldn't have started that bs sentence, only gets worse from here:

> and is considered to be a mental disease,

total BS, ICD-11 has just redefined it as not a mental but physiological disease by removing "transsexualism" and implementing "gender incongruence".

> it is as yet untreatable,

yeah suuuuuure, that's why I'm currently undergoing treatment -_-

> even for people who undergone surgery to remove their body parts the suicide rate is extremely high.

well, I'll grant you that one, but don't really see how it matters in this case. except you want your safe space to be exclusive for people not requiring any safe space ;)

> For 99.99% of population sex is immutable trait which determines gender.

while that number is reeeeeally nice and big and almost 100% it's also simply not true. `100-14 = 86 != 99.99`. see above.

> I am not going to be "inclusive" for the sake of ability to effectively filter my users.

ooooh, ok, yeah, can do that, your decision, but that's ableism, transphobia and bullshit ;)

at least those users who'd still use your service then will be as immoral as you, so no problem there.

> Second, are you seriously claiming artificialy altering one's skin colour makes one a black/white/whatever person?

of course, because IF you're dividing people by skin color (which was your idea, not mine), you should look at their skin, not their <s>sperm</s> genes.

> Furthermore, no, you can not change your sexual organs. You can have surgery to remove the organs you were born with, and have the surgeons remodel some parts of what remains to resemble organs of the opposite sex.

yeah, exactly that's called changing organs. but fun fact: you don't even need surgery for that, even hormone therapy starts to change the sexual organs.

> Third, you are religiously claiming to not allow anyone in is racism.

no, I'm claiming only allowing black people is racism. because that's what it is. see your favorite dictionary or read the first paragraph of https://en.wikipedia.org/wiki/Racism. even your "blacks only safe space based on biological differences" is mentioned in one of the first sentences.

> This is demonstrably false.

I like how you say big numbers and "demonstrably false" without any backing. Because your `99.99%` above were totally invented, I'd assume this claim to be too, unless you start to, well, demonstrate how it isn't.

> Sometimes having safe space requires exclusion of the people who would introduce hostile environment, or are not the target demographics.

that's true. but I doubt black underprivileged trans women are introducing hostile environments. or black underprivileged women with XY chromosome defects.

> When I am setting up website for black underprivileged women, I am setting it for them not for everyone.

understandable, but chromosomes are not the way to go, because they don't tell you anything valuable.

> > because genetics don't say anything about the person
> 
> We have come a long since the discovery of the DNA, and here you grandly proclaim "it does not say anything about the person".

true, because it doesn't. you want to prevent trolls and nazis from using your service, and you'll be unable to use DNA for checking that.

> Every physical trait has its reflection in the genome

not every, but most, yes, but when trying to filter away bad guys, physical traits don't tell you much ;)

> There is an ongoing research which shows links between DNA and many behavioral traits.

there are some traits that could be more or less accurately derived from a dna profile, but with strong emphasis on "some" and "less accurately". you'll be filtering some "mental diseases", but that's still just ableism and intolerant and will get you many false positives.

> > also, afaik, 23am tests cost about $100, I'd really like to see the shitstorm when you tell your underprivileged users to pay that just to use your website.
> 
> We are considering obtaining government funding from one of the many programs targeted at minorities.

if you're going to get funding for your program (which I like btw) I really hope you're going to spend it also on a way to be able to talk to those which were blocked by your DNA fence and determine if they were false positives.

but I doubt you'll get it, because "programs target at minorities" shouldn't fund people who say they're "not going to be inclusive because" they're too lazy to do any real profiling based on the personality of their users instead of DNA imaging.

DNA based auth is a good thing, but DNA based demography is not.

oh and watch http://www.imdb.com/title/tt0119177/ to see why depending on DNA profiling only is the wrong thing to do.

to be clear: while it's kinda sad that our society needs such safe spaces, I understand that it does and I like the idea of you making one. I'm not telling you not to do so or to allow anyone in. I'm just telling you that while DNA filtering will do most of the work for you, you should not get lazy over that (it's tempting to let the lab do all the work) but still allow people who were falsely excluded to get e.g. a face to face check to still get in. but saying "they're so few, I'll ignore them" is just the wrong thing to do.

**Comment by ethanwillis** [comment]
@nonchip  That prevalence section said that the transgender population is GROWING at 14% per year. The actual prevalence is much less.

**Comment by nonchip** [comment]
oh, thanks @sn3twork, my fault there. but still, we're too many to just ignore us, especcially for a project trying to do something good for minorities. (better numbers: http://www.gendercentre.org.au/resources/polare-archive/archived-articles/how-many-of-us-are-there.htm says about 1:30k-1:100k in some european studies; and that are just those who've already undergone or are actively seeking surgery)

it's just that @blackfrat's argumentation seems pretty bigot by saying they're doing a project to support minorities but don't want to be "too inclusive" just because of "some minor statistical errors". that's the same argumentation as people who "are no nazis [so they claim], but don't like minorities" in their project/town/whatever. or actually even worse, because it sounds like they're just too lazy. also actively ignoring the problem as the "please disregard this complaint" statement shows. it's kind of "out of sight, out of mind"…

**Comment by Matsurishi** [comment]
Wow! Thank you so much! Can't wait to set up my whites-only safe space and exclude all niggers! I fucking love you!

**Comment by ethanwillis** [comment]
@nonchip Well there are genetic bases for gender dysphoria as well. The analysis might not yet be up to par, but this could be modified/extended to also include trans people.

**Comment by Matsurishi** [comment]
I really am sick of all those filthy niggers shitting up all of our glorious white communities on the internet! 4chan will be really quite grateful for this one, thank you!

**Comment by nonchip** [comment]
@sn3twork it could be modified to include those trans people who are trans because of genetics, but not all the others. it's true there are some genetic bases for gender dysphoria, but that doesn't mean all gender dysphoria is genetically based. so yes, the system can always get better, but never good enough to become the only option when applied to individual people it wasn't trained for.

@Matsurishi please get your account closed and sued for the sick bullshit you're writing, thanks.

**Comment by ethanwillis** [comment]
@nonchip That is true, this system isn't perfect. At the same time I don't think the author believes it or tries to pass it off that it's perfect. I do think there's some value here for quick categorization of individuals who have the genetic basis for being quickly identified as true positives. If this system is used in that way then it reduces management overhead when creating safe spaces. And I think that's the main takeaway?

 This system augmented with alternative methods of more manual identification and systems to allow individuals to ask for a human review would be better obviously. I just think those systems fall outside the scope of an OAuth project.

**Comment by ganeshran** [comment]
Regardless of what the author's intentions are, tools like these have massive potential for misuse. Imagine having an entire section of the internet walled off simply by virtue of your race/gender etc.

Anyways, more than the API author, it is the provider of genetic data (23andme) which should be more concerned about misuse. The gattaca-ish implications of the tool are not lost on most people.

**Comment by ethanwillis** [comment]
@ganeshran Lots of tools we produce as developers have massive potential for misuse. However they also have massive potential for good uses as well. That's where as users and downstream developers our responsibility for how we use the tools we're given comes in.

Also, the provider of the data shouldn't be concerned about the misuse because the users are the ones who control their own data and decide when and how it is used.

**Comment by ethanwillis** [comment]
And before someone chimes in saying how easy it is for me to say these things as a white male, I'd like to say that in fact I'm not. I'm 1/4th "asian" and wouldn't be passing any racial purity tests on either side if some kind of racist walled garden were built.

I can let anyone see my 23andme data to back that up too, and only if I expressly authorize 23andme to share my data!

**Comment by blackfrat** [comment]
> tal BS, ICD-11 has just redefined it as not a mental but physiological disease by removing "transsexualism" and implementing "gender incongruence"

A lie. Your condition is categorized as ICD-10. 
The fact that "World Professional Association for Transgender Health" _recommended_ [Obviously because they have no idea how to treat it, except through bodily mutilation] it to be removed in 2017 does not change that. 

> yeah, exactly that's called changing organs. but fun fact: you don't even need surgery for that, even hormone therapy starts to change the sexual organs.

This is laughable, surgicaly molding skin is not "changing sexual organs". I wait until you declare yourself lionkin after surgically attaching great crest onto your head. You would be as much lion, as you would be man/woman after surgery. Furthermore, regardless of the "identity" you will identify with, if you try to pass as a person with surgicaly crafted "organs", I imagine your hypothetical SO won't be happy with you having no womb, or no testicles. Unless you will want to pose as a "woman" with testicles, or "man" with a womb. So much for "treatment".
I would be pretty pissed off, when after dating for 5 years I discover my SO has no actual testicles. So if you are responsible, you would need to let your SO know this, which contradicts your claims you would "become" male/female.  

> yeah, sure, 14% of the population is "extremely rare":

This is, of course, a lie. 
We would be doomed as a species when 14% percent of populace would be born with genetic defects or develop mental illness later in life. 

Transsexuals are about  0.5% at most. The wiki you cited even confirms this. Yes I know, statistics is bigotry, and feelz are science. 

The bottom line is, you would like this project to be dropped because of your personal reasons, at the same time forcing me to admit painting your face black means you are a black person, and cutting testicles off means male becomes female, or no, just wearing a make up makes you female. This is anti-scientific bigotry feeding on emotions, nothing more.

**Comment by nonchip** [comment]
@blackfrat:

> A lie. Your condition is categorized as ICD-10.

A stupid. see how I explicitely wrote ICD-11. which is currently draft and contains the mentioned changes, not just some group recommending something.

> Obviously because they have no idea how to treat it, except through bodily mutilation

well, what you call "bodily mutilation" is what I call changing my body to match my personality. you know, because I, as a member of the self aware species of humanity, am what I am, not what my body is. because if I lost a leg in an accident tomorrow, I'd still want my leg back, and you'd give me a prosthetic, not therapy to make me feel like having just one leg is better.

> This is laughable, surgicaly molding skin is not "changing sexual organs".

that's why I told you you don't need surgery. try to read, maybe?

> Furthermore, regardless of the "identity" you will identify with, if you try to pass as a person with surgicaly crafted "organs", I imagine your hypothetical SO won't be happy with you having no womb, or no testicles. Unless you will want to pose as a "woman" with testicles, or "man" with a womb. So much for "treatment".
> I would be pretty pissed off, when after dating for 5 years I discover my SO has no actual testicles. So if you are responsible, you would need to let your SO know this, which contradicts your claims you would "become" male/female.

congrats, you just managed to demonstrate your stupidity, because:
- I have a (non-hypothetical) boyfriend
- he knows my body pretty well and even knew it before we came together, but because he's not such a transphobic, shallow person as you, he likes me for who I am, not for the womb I don't have
- you seriously think I'd date someone for 5 years before they "discover my sexual organs"? it's not 1500 anymore :P
- of course I told him I have the opposite sexual organs, but that doesn't contradict anything, because he e.g. reminds me of my daily hormone dose, supports me emotionally on my way, etc.

again: I don't want to hide by "fitting into the role", i just want my body to not depress me by looking like I fit into the OTHER role.

me telling a stranger on a public github issue thread all these details should also show how I don't try to hide.

> This is, of course, a lie. 

congrats on pointing out what others already did: I misread and I apologied for that.

> We would be doomed as a species when 14% percent of populace would be born with genetic defects or develop mental illness later in life.

well, then we're doomed, see e.g. http://www.thekimfoundation.org/html/about_mental_ill/statistics.html or just google for "percentage of mental illnesses".

> Transsexuals are about 0.5% at most. The wiki you cited even confirms this.

I know. I misread. see above

> Yes I know, statistics is bigotry, and feelz are science.

I never said that. and again: see above.

> The bottom line is, you would like this project to be dropped because of your personal reasons,

a lie. i never said that. i just said I'd like the claim of this project "creating safe spaces" to be dropped, or at least a note added reminding people like you how important it is to have a second option, because DNA isn't accurate enough. for reference please maybe actually READ MY POSTS?

> at the same time forcing me to admit painting your face black means you are a black person,

no, I don't. I told you most people agree e.g. Michael Jackson was white, because that's how he looked. and especially with safe spaces for people who might be oppressed because they are black, maybe you want to look at their skin, not their genes, because that's what the racists you want to protect them from do too. the nazi gangs don't do dna tests, so why do you want to depend on them when protecting people from them?

> and cutting testicles off means male becomes female,

again, I never said that. I just said you should stop insulting me just because I want to get rid of the male parts that don't belong to me cause I already am and always was female, despite having them because my body developed wrong.

> or no, just wearing a make up makes you female.

now you're getting really stupid. wearing makeup makes you wear makeup, nothing else. and I never claimed anything else.

> This is anti-scientific bigotry feeding on emotions, nothing more.

_yawn_ true, because you invented that whole "bottom line", being either too lazy or too stupid to actually read my arguments before flaming against what you thought i might have meant.

**Comment by blackfrat** [comment]
Do you know what a draft is? It means the _official_ state, is you suffer from the condition categorized as ICD-10. Maybe in 2017 you will magically be "treated" after they redefine the category. But for now, it is not, and you suffer you ICD-10 condition. 

> that's why I told you you don't need surgery. try to read, maybe?

Taking certain drugs might give you something akin to female breasts. Sexual organs, though, don't grow on trees, so you need artificial substitute which will be disfunctional, ie. won't allow for reproduction.  

> or just google for "percentage of mental illnesses".

No, I meant this particular one. 

> despite having them because my body developed wrong.

Or maybe you suffer from gender identity disorder ICD-10:

"The desire to live and be accepted as a member of the opposite sex, usually accompanied by the wish to make his or her body as congruent as possible with the preferred sex through surgery and hormone treatment"
Unless you performed genetic tests to verify that indeed you are woman without Y chromosome, which I suspect you did not, because then you would not suffer from gender identity disorder, because you would objectively be a female. 

> I want to get rid of the male parts that don't belong to me cause I already am and always was female, despite having them because my body developed wrong.

"I was always a female, except that I need to cut off my genitals to finally become one"
and
"I was always rich, except when I was poor, because my purse developed wrong"

are both nonsensical sentences.

> no, I don't. I told you most people agree e.g. Michael Jackson was white, 

Michael Jackson was a black man who through chemical treatment lost his pigment. This why I am "afraid" of people like you. You are living in alternate reality, pushing your insanity on others as in:
- Man after taking hormones and undergoing private part removals becomes a female. ("transsexual" or "transgender") 
- After attaching a crest I become a lion. My DNA clearly developed wrong. ("otherkin")
- After wearing a makeup white woman becomes black (it is called "transracial", and is becoming a real thing. Glimpse into the insanity: google Rachel Dolezal.
- Wearing makeup does make you a female: see Bruce Jenner. Who did not even bother to have his private parts removed, which served him well, as he fathered a handful of children. 

Now, excuse me as I pickup my 51 lost gender.

**Comment by nonchip** [comment]
> Do you know what a draft is? It means the official state…

… is considered outdated and needs updating, because else you wouldn't pay people to draft the update. and stop telling me what I'm suffering from, thanks, I know that myself.

> Taking certain drugs might give you something akin to female breasts.

if by "something akin" you mean "the exact same", yes. some trans women even start to lactate from the hormone therapy

> Sexual organs, though, don't grow on trees

no, but on people. that's why I'm taking the meds, not giving them to trees ;)

> so you need artificial substitute

wrong, cause (as you would know if you read my previous statements), the meds already started to change the functions of my primary sexual organs, and also: don't tell me what I need.

also this is completely irrelevant, as you might have noticed this issue is about the problems of misimplementing DNA based access control, not about what you might think you know about my specific condition.

> which will be disfunctional

wrong, I can have sex with them perfectly fine ; see below

> ie. won't allow for reproduction.

that's not the function I intend to use them for anyway, cause I don't want to have kids ; see above

> No, I meant this particular one.

um, no, that's not what you wrote, sorry. also, you said "mental illness", which my condition is not, as I clarified before.

> Or maybe you suffer from gender identity disorder ICD-10

or maybe that's what I already confirmed when saying that ICD-11 is dropping that classification.

> Unless you performed genetic tests to verify that indeed you are woman without Y chromosome, 

which I did, turned out that i am indeed not, and that I never claimed to be.

> which I suspect you did not

oops, sorry to burst your bubble

> because then you would not suffer from gender identity disorder,

well, while the one condition doesn't exclude the other, it's true that i only have one of them

> because you would objectively be a female.

I feel really sorry for you. You seriously think some miscopied DNA is more "objectively true" than the consciousness of a person? didn't read that part about humans being more or (in your case) less intelligent creatures, did you?

also, again, please focus on the topic, not on your misconception of being able to teach me my own body and mind, because you are definitely not. thanks.

---

> [everything after the totally forged quote that is true despite you being too narrow to understand how that has nothing to do with being rich/poor or a lion or wtf else]

I feel really sorry for you. Seriously, I feel sorry for you. you are afraid of your own species not because it's the only one capable of destroying a whole planet many times just by one of its members deciding to do so, but because of its members you consider dangerous because we "are pushing your insanity" just by looking forward to be accepted. that's making me feel really sad and sorry for you.

btw your lion crest example is total BS and you know it.
also, the first name of your 4th example is "Caitlyn", not "Bruce". oh and she had kids before she knew (or at least publicly announced) she's a woman.

oh and one final advice: PLEASE read into the sex vs gender issue. your argumentation strongly suggests you just don't know the fact that sexual organs have little to do with (sozio-)psychological gender.

also: get out of your home, drive to your nearest bigger city and just look at the people. see how they are all individuals? of course you'll see some white guys in suits (who are often stereotyped as "normal" by our society), but what about everyone else? humans are individuals who should be respected for their individuality, because that's what makes our minds work. srsly, try to open your mind a bit more and look if it feels better or worse. personally, I feel better when interacting with open minded people, and also when being open minded while interacting with them.

**Comment by halbekanne** [comment]
I have to say, it's hilarious to see different SJWs fight over how this inherently sexist, racist and transphobic software should be used in order to not effectively use them against those sexes or races they prefer. If you ask me, this is a pretty impossible task. You filter users through DNA? Hilter whished he had this kind of software to create a "safe space" for anyone who is/isn't Aryan. He still had to use old methods.

![](http://www.circe-film.de/produktion/liebekrieg/img/besatz4.jpg)

Also, can't I just use this software to e.g. create a safe space for white cis males?

Honestly, I don't think the developer thought this through.

Edit: "This app clearly violates our API policy," a spokesperson said. "We've shut down the application and this developer no longer has access to our API... Using our API to develop applications that contain, display or promote hate is prohibited by our API Terms of Service."

The developer really didn't thought this through.

**Comment by nonchip** [comment]
thanks, @MoonShade, for bringing this back on topic, and for summarizing my initial statement pretty well.
